proxy/http2/Http2ConnectionState.cc

```diff
@@ -936,30 +940,70 @@ Http2ConnectionState::cleanup_streams()
 void
 Http2ConnectionState::delete_stream(Http2Stream *stream)
 {
+  // The following check allows the method to be called safely on already deleted streams.
+  if (deleted_from_active_streams(stream)) {
+    return;
+  }
+
+  SCOPED_MUTEX_LOCK(lock, this->mutex, this_ethread());
+
```

Comment:
Why do you need to grab a lock here?  Given the mutex sharing between vc and session and ConnectionState, we should already be holding this mutex.